examples/basic/example_test.go

```diff
@@ -47,11 +47,20 @@ func ExampleBasicPlugin() {
 
 	for i := 0; i < numNodes; i++ {
 		builder := network.NewBuilder()
-		builder.SetKeys(ed25519.RandomKeyPair())
-		builder.SetAddress(network.FormatAddress("tcp", host, uint16(startPort+i)))
+		address := network.FormatAddress("tcp", host, uint16(startPort+i))
 
-		builder.AddPlugin(new(discovery.Plugin))
+		// enforce S/Kademlia node IDs
+		plugin := discovery.New(
+			discovery.WithPuzzleEnabled(discovery.DefaultC1, discovery.DefaultC2),
+		)
+		kp, nonce := plugin.PerformPuzzle()
 
+		builder.SetKeys(kp)
+		id := peer.CreateID(address, kp.PublicKey)
```

Comment:
Care to modify peer.CreateID to support options s.t. `peer.CreateID(address, kp.PublicKey, peer.WithNonce(nonce), ...otherOptions)`?